templates/terraform/resource_iam.html.markdown.erb

```diff
@@ -0,0 +1,156 @@
+<%- # the license inside this block applies to this file
+# Copyright 2017 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+-%>
+<%# NOTE NOTE NOTE
+    The newlines in this file are *load bearing*.  This file outputs
+    Markdown, which is extremely sensitive to newlines.  You have got
+    to have a newline after every attribute and property, because
+    otherwise MD will think the next element is part of the previous
+    property's bullet point.  You cannot have any double newlines in the
+    middle of a property or attribute, because MD will think that the
+    empty line ends the bullet point and the indentation will be off.
+    You must have a newline before and after all --- document indicators,
+    and you must have a newline before and after all - - - hlines.
+    You cannot have more than one blank line between properties.
+    The --- document indicator must be the first line of the file.
+    As long as you only use `build_property_documentation`, it all works
+    fine - but when you need to add custom docs (notes, etc), you need
+    to remember these things.
+
+    Know also that the `lines` function in heavy use in MagicModules will
+    strip exactly one trailing newline - unless that's what you've designed
+    your docstring for, it's easier to insert newlines where you need them
+    manually.  That's why, in this file, we use `lines` on anything which
+    is generated from a ruby function, but skip it on anything that is
+    directly inserted from YAML.
+-%>
+<%
+  if @config.legacy_name.nil?
+    terraform_name = "google_" + (product_ns + object.name).underscore + "_iam"
+  else
+    terraform_name = "google_" + @config.legacy_name + '_' + object.name.underscore + "_iam"
+  end
+  properties = object.all_user_properties
+  timeouts = object.async.nil? ? Api::Timeouts.new : object.timeouts
+-%>
+---
+<%= lines(autogen_notice :yaml) -%>
+layout: "google"
+page_title: "Google: <%= terraform_name -%>"
+sidebar_current: "docs-<%= terraform_name.gsub("_", "-") -%>"
+description: |-
+  Collection of resources to manage IAM policy for <%= product_ns -%> <%= object.name -%>
+
+Three different resources help you manage your IAM policy for <%= product_ns -%> <%= object.name -%>. Each of these resources serves a different use case:
```

Comment:
If we can construct reasonable human-readable names, should we preserve the title header? https://github.com/terraform-providers/terraform-provider-google-beta/pull/914/files#diff-5bc4b7188e596ed679a95ddfd0c495c2L9

Reply:
I was a little confused on this because for most resources we use `google_resource_name` as the title header, but IAM has a different pattern? Can add it back

Reply:
I think we just use a different pattern because IAM resources are 3 resources grouped in a single page. I'd like to keep *a* header but if you have a better idea of what to write there that's good with me as well.